operator/internal/component/podclique/pod/syncflow.go

```diff
@@ -232,6 +253,69 @@ func (r _resource) checkAndRemovePodSchedulingGates(sc *syncContext, logger logr
 	return skippedScheduleGatedPods, nil
 }
 
+// isIndividualPodGang determines if a PodGang is an individual PodGang (beyond MinAvailable).
+// Individual PodGangs have names like: "simple1-0-sga-0", "simple1-0-sga-1", etc.
+func isIndividualPodGang(podGangName string) bool {
+	// Individual PodGangs contain the pattern "-sga-" followed by a 0-based index
+	// This is based on the naming convention: {pgsName}-{pgsReplica}-{scalingGroupName}-{individualPodGangIndex}
+	return strings.Contains(podGangName, "-sga-") && strings.Count(podGangName, "-") >= 3
+}
+
+// extractBasePodGangName extracts the base PodGang name from an individual PodGang name.
+// Example: "simple1-0-sga-0" -> "simple1-0"
+func extractBasePodGangName(individualPodGangName string) string {
+	// Find the scaling group pattern "-sga-"
+	sgaIndex := strings.Index(individualPodGangName, "-sga-")
+	if sgaIndex == -1 {
+		// This shouldn't happen for individual PodGangs, but return the original name as fallback
+		return individualPodGangName
+	}
+
+	// Return everything before "-sga-"
+	return individualPodGangName[:sgaIndex]
+}
+
+// isBasePodGangReady checks if all PodCliques in the base PodGang meet their MinAvailable requirements.
+func isBasePodGangReady(ctx context.Context, c client.Client, logger logr.Logger, namespace, individualPodGangName string) bool {
+	basePodGangName := extractBasePodGangName(individualPodGangName)
+
+	// Get the base PodGang
+	basePodGang := &groveschedulerv1alpha1.PodGang{}
+	basePodGangKey := client.ObjectKey{Name: basePodGangName, Namespace: namespace}
+
+	if err := c.Get(ctx, basePodGangKey, basePodGang); err != nil {
```

Comment:
can you change the name of the existing method `getAssociatedPodGang` and just call it `getPodGang` and reuse this here and in `prepareSyncFlow`. This way we prevent duplicate code.